launcher/util.go

```diff
@@ -28,3 +31,20 @@ func FetchImpersonatedToken(ctx context.Context, serviceAccount string, audience
 
 	return []byte(token.AccessToken), nil
 }
+
+func listFilesWithPrefix(targetDir string, prefix string) ([]string, error) {
+	var targetFiles []string
+	err := filepath.WalkDir(targetDir, func(path string, d os.DirEntry, err error) error {
+		if err != nil && d != nil && d.IsDir() {
+			return filepath.SkipDir
+		}
+		if d != nil && !d.IsDir() && strings.HasPrefix(filepath.Base(path), prefix) {
```

Comment:
In what circumstances can err be non-nil and this path still be valid? It worries me that we're skipping a non-nil err check